File: BOGUdanyo/src/main/java/com/backend3rd/BOGUdanyo/api/KakaoRestApiHelper.java
package com.backend3rd.BOGUdanyo.api;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;
import org.springframework.stereotype.Component;
import org.springframework.beans.factory.annotation.Value;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.UnsupportedEncodingException;
import java.net.MalformedURLException;
import java.net.URL;
import java.net.URLConnection;
import java.net.URLEncoder;

@Component
public class KakaoRestApiHelper {
    @Value("${KAKAO_API_KEY}")
    String apiKey;
    public float [] getKakaoApiFromAddress(String roadFullAddr) throws ParseException {
        String apiUrl = "https://dapi.kakao.com/v2/local/search/address.json";
        String jsonString = null;
        float [] coordi;

        try {
            roadFullAddr = URLEncoder.encode(roadFullAddr, "UTF-8");

            String addr = apiUrl + "?query=" + roadFullAddr;

            URL url = new URL(addr);
            URLConnection conn = url.openConnection();
            conn.setRequestProperty("Authorization", "KakaoAK " + apiKey);

            BufferedReader rd = null;
            rd = new BufferedReader(new InputStreamReader(conn.getInputStream(), "UTF-8"));
            StringBuffer docJson = new StringBuffer();

            String line;

            while ((line=rd.readLine()) != null) {
                docJson.append(line);
            }

            jsonString = docJson.toString();
            rd.close();

        } catch (UnsupportedEncodingException e) {
            e.printStackTrace();
        } catch (MalformedURLException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }

        coordi = changeToJSON(jsonString);
        return coordi;
    }

    private float[] changeToJSON(String jsonString) throws ParseException {

        JSONParser parser = new JSONParser();
        JSONObject document = (JSONObject) parser.parse(jsonString);
        JSONArray jsonArray = (JSONArray) document.get("documents");
        JSONObject position = (JSONObject) jsonArray.get(0);
        float x = Float.parseFloat((String) position.get("x"));
        float y = Float.parseFloat((String) position.get("y"));

        return new float[]{x, y};

    }

}

File: BOGUdanyoKT/src/main/kotlin/com/backend3rd/BOGUdanyoKT/api/KakaoRestApiHelper.kt
package com.backend3rd.BOGUdanyoKT.api

import org.json.simple.JSONArray
import org.json.simple.JSONObject
import org.json.simple.parser.JSONParser
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import java.io.IOException
import java.io.UnsupportedEncodingException
import java.net.HttpURLConnection
import java.net.MalformedURLException
import java.net.URL
import java.net.URLEncoder

@Component
class KakaoRestApiHelper (
        @Value("\${kakao_api_key}")
        val apiKey: String
        ){

    fun getKakaoApiFromAddress(roadFullAddr: String): Array<Float> {
        val apiUrl = "https://dapi.kakao.com/v2/local/search/address.json?query="
        var jsonString = ""
        var cordi: Array<Float> = arrayOf(0F, 0F)
        try {
            val address = URLEncoder.encode(roadFullAddr, "UTF-8")

            val obj = URL(apiUrl + address)

            val con: HttpURLConnection = obj.openConnection() as HttpURLConnection

            con.requestMethod = "GET"
            con.setRequestProperty("Authorization", "KakaoAK " + apiKey)
            con.setRequestProperty("content-type", "application/json")
            con.doOutput = true
            con.useCaches = false
            con.defaultUseCaches = false

            val rd = con.inputStream.bufferedReader()
            var docJson = StringBuffer()

            while(true){
                var line = rd.readLine()
                if(line == null) break
                docJson.append(line)
            }

            jsonString = docJson.toString()
            rd.close()
        }
        catch (e : UnsupportedEncodingException){
            e.printStackTrace()
        }
        catch (e: MalformedURLException){
            e.printStackTrace()
        }
        catch (e: IOException){
            e.printStackTrace()
        }
        catch (e: Exception){
            e.printStackTrace()
        }

        cordi = changeToJSON(jsonString)
        return cordi
    }

    fun changeToJSON(jsonString: String): Array<Float>{
        val parser = JSONParser()
        val document = parser.parse(jsonString) as JSONObject
        val jsonArray = document["documents"] as JSONArray
        val position = jsonArray[0] as JSONObject
        val x = position["x"].toString().toFloat()
        val y = position["y"].toString().toFloat()

        return arrayOf(x, y)
    }
}